task12/index.js

```diff
@@ -0,0 +1,56 @@
+const parent = document.getElementById("js-parent");
+    
+const appendButtonForParent = () => {
+    const button = document.createElement('input');
+    button.id = "button";
+    button.type = "button";
+    button.value = "click";
+    parent.appendChild(button);
+};
+appendButtonForParent();
+
+const removeButton = () => {
+    document.getElementById("button").remove();
+};
+
+const addEvent = document.getElementById("button"); 
+addEvent.addEventListener('click', () => {
+    removeButton();
+    init();
+});
+
+const getData = async () => {
+    try {
+        const request = await fetch("https://jsondata.okiba.me/v1/json/3jhPW210812052923");
+        const data = request.json();
+        return data;
+            
+    } catch (error) {
+        parent.textContent = "データを読み込めませんでした";
+        console.error(error);
+    }
+};
+
+const appendListForParent = (value) => {
```

Comment:
IMO：

When there is an error in the URL passed in the fetch, I think it goes through the error clause.
After that, I noticed that there was a program error when the appendListForParent function was passed.

![スクリーンショット 2021-09-09 11 32 41](https://user-images.githubusercontent.com/60378833/132612722-235bbd93-9e91-4408-a21b-8c892e7b7cc2.png)


Therefore, I felt it necessary to handle the condition when a falthy value is entered in the value in the appendListForParent function.

Reply:
It was a part I didn't notice! You're right!
Introduced try / catch in the init function to enable error handling.
 Thank you for pointing out.

Reply:
I'm sorry. I was not very good at communicating.
This is not an intended correction.

Let me be more specific.
Provisionally, After passing through the error clause, I think accessing the argument value that was contains the variables of the data in the `appendListForParent` function, and at that time, the value should be undefiend, and I think a program error will occur at the same time.
※It means about the first image I sent as an attachment.

So, I thought it would be smarter to add error handling when there is nothing in value to the `appendListForParent` function.

![スクリーンショット 2021-09-09 23 43 45](https://user-images.githubusercontent.com/60378833/132707365-dd844999-a72b-44bf-9d81-66138f657976.png)

There is no need to add try/catch to the init function.